controllers/apps/operations/rebuild_instance.go

```diff
@@ -91,38 +73,92 @@ func (r rebuildInstanceOpsHandler) ActionStartedCondition(reqCtx intctrlutil.Req
 }
 
 func (r rebuildInstanceOpsHandler) Action(reqCtx intctrlutil.RequestCtx, cli client.Client, opsRes *OpsResource) error {
-	if opsRes.OpsRequest.Spec.Force {
-		return nil
-	}
 	for _, v := range opsRes.OpsRequest.Spec.RebuildFrom {
 		compStatus, ok := opsRes.Cluster.Status.Components[v.ComponentName]
 		if !ok {
 			continue
 		}
 		// check if the component has matched the `Phase` condition
-		if !slices.Contains([]appsv1alpha1.ClusterComponentPhase{appsv1alpha1.FailedClusterCompPhase,
+		if !opsRes.OpsRequest.Spec.Force && !slices.Contains([]appsv1alpha1.ClusterComponentPhase{appsv1alpha1.FailedClusterCompPhase,
 			appsv1alpha1.AbnormalClusterCompPhase, appsv1alpha1.UpdatingClusterCompPhase}, compStatus.Phase) {
 			return intctrlutil.NewFatalError(fmt.Sprintf(`the phase of component "%s" can not be %s`, v.ComponentName, compStatus.Phase))
 		}
+		var (
+			synthesizedComp *component.SynthesizedComponent
+			err             error
+			instanceNames   []string
+		)
 		for _, ins := range v.Instances {
```

Comment:
Shall we limit the RebuildInstance API to specify only one instance could simplify the implementation?
Additionally, we could also restrict the RebuildFrom []RebuildInstance slice to contain at most one element.

I think this approach would reduce code complexity.

Reply:
btw, even if end-user would like to rebuild multiple instances, our implementation could be executing it serially (one by one).

Reply:
good idea. i will add the policy in future